docs/how_to/use_estimator.rst

```diff
@@ -0,0 +1,145 @@
+###########################
+Use the Estimator primitive
+###########################
+
+This guide shows how to get the expected value of an observable for a given quantum circuit with the :class:`~qiskit.primitives.Estimator` primitive.
+
+.. note::
+
+    While this guide only uses Qiskit Terra's implementation of the ``Estimator`` primitive, there are other
+    implementations of this primitive like Qiskit Terra's :class:`~qiskit.primitives.BackendEstimator`, Qiskit Aer's :class:`~qiskit_aer.primitives.Estimator`
+    and Qiskit Runtime's :class:`~qiskit_ibm_runtime.Estimator`.
+
+
+Initialize observable
+=====================
+
+The first step is to define the observable whose expected value you want to compute. This observable can be any ``BaseOperator``, like the operators from :mod:`qiskit.quantum_info`.
+Among them it is preferable to use :class:`~qiskit.quantum_info.SparsePauliOp`.
+
+.. testcode::
+
+    from qiskit.quantum_info import SparsePauliOp
+
+    obs = SparsePauliOp(["II", "XX", "YY", "ZZ"], [1, 1, -1, 1])
+
+Initialize quantum circuits
+===========================
+
+Then you need to create the :class:`~qiskit.circuit.QuantumCircuit` for which you want to obtain the expected value.
+
+.. plot::
+    :include-source:
+
+    from qiskit import QuantumCircuit
+
+    qc = QuantumCircuit(2)
+    qc.h(0)
+    qc.cx(0,1)
+    qc.draw("mpl")
+
+.. testsetup::
+
+    from qiskit import QuantumCircuit
+
+    qc = QuantumCircuit(2)
+    qc.h(0)
+    qc.cx(0,1)
+
+Initialize the ``Estimator``
+============================
+
+Then, you need to create an :class:`~qiskit.primitives.Estimator` object.
+
+.. testcode::
+
+    from qiskit.primitives import Estimator
+
+    estimator = Estimator()
```

Comment:
By default this gives an ideal result (akin to using statevector) - should we mention this?

Reply:
I'm not sure this is the place for that. This is supposed to be only a step-by-step guide for working with an `Estimator` instead of an explanation about the differences between implementations or going into all the details of the class (this is what API reference is for).  I guess it depends on how important/useful we consider to tell people how to configure the `shots` in `options` here.

Moreover, we plan to write an explanation about the primitives soon. It might be better to cover that there. A tutorial is also a fine place to mention that.

Reply:
For me since this all around an Estimator configuring it (since it can work differently) to do the job you need seems highly relevant. I would not expect any of that for the QuantumCircuit or Operator it uses but since howto for using the Estimator not saying anything I am just not sure. Though I would not say much that's for certain - which is why I said mention not explain. Part of my rationale for mentioning is that I have had to answer issues where 'its my code doesn't give the same result/fails' when going from an qiskit primitives Estimator() to an Aer Estimator() both just constructed as defaults. Knowing what it does/how its configured is key there. Since there is a note pointing to the other estimators what is the implication - they are the same and can be swapped in. I have had to give people explanation on more than one occasion. I don't think this howto should deal with that of course, but maybe it should state at least how the Estimator will behave in the config its used in.

Reply:
If it's a common user pain point I guess it might be better to mention that explicitly. 

Now I'm also thinking about adding an example in which I change from the default `Statevector`-based primitives to shot-based so a user can see quickly how to set the `options` argument in the primitives, as it's not obvious if you just look at the API reference.

Reply:
There are two ways there if you did not figure. It can be set when calling run()  - via kwargs (named run_options in code if you end up looking) - so you can change it on a call by call basis. The value used on run() can be also be set via the constructor - options - for all run invocations (overridden by the one if passed to run). This latter via options is more normal if a user wants to use an Estimator with an algorithm like VQE where you want as a user to config the shots used in circuit execution later when used by VQE. Since the howto is more about using an Estimator and calling run yourself maybe the shots on run() is more suitable to an example here. You can see examples of both in the unit tests here https://github.com/Qiskit/qiskit-terra/blob/fa7ac688fdce3ecc27fb7e13d8b2c3f8553b3ff8/test/python/primitives/test_estimator.py#L612-L633